### PR TITLE
Adding the edittext factory tests

### PR DESCRIPTION
**Pull request by dubdabasoduba**

### CODE REVIEW

File: android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java [hunk 1]
```diff
@@ -412,7 +416,7 @@ private void attachRefreshLogic(Context context, JSONObject jsonObject, Material
         }
     }
 
-    private MaterialEditText getViewUsingAddress(String stepName, String fieldKey, JsonApi jsonApi) {
+    public MaterialEditText getViewUsingAddress(String stepName, String fieldKey, JsonApi jsonApi) {
```
Comment (ndegwamartin) [inline android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java:419]: I see we've altered the access modifier here, do we intend to override it on the  implementations?

Comment (dubdabasoduba) [reply android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java:419]: Its used in mocking on the tests

Comment (ndegwamartin) [reply android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java:419]: Ok, since the test is in a similar package you can give it a narrower scope and use **_protected_** instead. Also, annotate the method with **_@VisibleForTesting_**

Comment (dubdabasoduba) [reply android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java:419]: Yah this makes sense

File: android-json-form-wizard/src/test/java/com/vijay/jsonwizard/widgets/EditTextFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        Mockito.doReturn(relatedEditText2).when((JsonApi) context).getFormDataView("step1:medications_other2");
+        Mockito.doReturn(totalValueEditText).when((JsonApi) context).getFormDataView("step1:medications_other3");
+
+        String gpsString = "{\"key\":\"user_last_name\",\"openmrs_entity_parent\":\"\",\"openmrs_entity\":\"\",\"openmrs_entity_id\":\"\",\"type\":\"edit_text\",\"hint\":\"User Last name\",\"edit_type\":\"name\",\"value\":\"Lifted\",\"v_min_length\":{\"value\":\"6\",\"err\":\"Please enter a valid name\"},\"v_max_length\":{\"value\":\"10\",\"is_fixed_size\":\"true\",\"err\":\"Please enter a valid name\"},\"v_required\":{\"value\":\"true\",\"err\":\"Please enter the last name\"},\"v_cumulative_total\":{\"value\":\"medications_other3\",\"err\":\"Enter required fields\",\"related_fields\":[\"medications_other\",\"medications_other2\"]}}";
+        List<View> viewList = factorySpy.getViewsFromJson("step1", context, formFragment, new JSONObject(gpsString), listener);
+        Assert.assertNotNull(viewList);
+        Assert.assertTrue(viewList.size() > 0);
```
Comment (ndegwamartin) [inline android-json-form-wizard/src/test/java/com/vijay/jsonwizard/widgets/EditTextFactoryTest.java:215]: I think we are better off asserting the exact value we expect based on the input in order to cover the ceiling edge case e.g. no duplicates
